debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleDatabaseSchema.java

```diff
@@ -149,32 +150,15 @@ public TableId getTableIdByObjectId(Long objectId, Long dataObjectId) {
         // Internally we cache this using a bounded cache for performance reasons, particularly when a
         // transaction may refer to the same table for consecutive DML events. This avoids the need to
         // iterate the list of tables on each DML event observed.
-        return objectIdToTableId.computeIfAbsent(objectId, (tableObjectId) -> {
-            for (TableId tableId : tableIds()) {
-                final Table table = tableFor(tableId);
-                final Attribute attribute = table.attributeWithName(ATTRIBUTE_OBJECT_ID);
-                if (attribute != null && attribute.asLong().equals(tableObjectId)) {
-                    if (dataObjectId != null) {
-                        final Attribute dataAttribute = table.attributeWithName(ATTRIBUTE_DATA_OBJECT_ID);
-                        if (dataAttribute == null || !dataAttribute.asLong().equals(dataObjectId)) {
-                            // Did not match, continue
-                            continue;
-                        }
-                    }
-                    LOGGER.debug("Table lookup for object {} resolved to '{}'", tableObjectId, table.id());
-                    return table.id();
-                }
-            }
+        TableId cachedTableId = objectIdToTableId.putIfAbsent(objectId, tableObjectIdToTableId(objectId, dataObjectId));
```

Comment:
OK, `putIfAbsent` was not a good choice because right now I'd say we are back to square one, doing the expensive search in the `tableObjectIdToTableId` for something we might not need at all.
So in this regard the plain
```
var cachedTableId = objectIdToTableId.get(objectId);
if (cachedTableId == null) {
  cachedTableId = tableObjectIdToTableId(objectId, dataObjectId);
} 
```
would probably work best